src/app/tests/TestReadInteraction.cpp

```diff
@@ -1748,6 +1748,7 @@ void TestReadInteraction::TestSubscribeWildcard(nlTestSuite * apSuite, void * ap
 
         delegate.mGotReport = false;
 
+        attributePathParams.release();
         err = readClient.SendAutoResubscribeRequest(std::move(readPrepareParams));
```

Comment:
Our usual policy has been that until this call succeeds, the onus to free the client object is on the caller. It seems like de-allocation of paths doesn't quite line up that way, and the logic within `SendAutoResubscribeRequest` takes responsibility for calling you to free up the params. I realize this is existing code, but perhaps we line those up?

Reply:
The reason it works like it does is that once you call `std::move` on something you're not really supposed to assume things about its state.  Any cleanup should really be done by the callee at that point....

Reply:
Why not `std::move` it back if any errors happen within `SendAutoResubscribeRequest`?

Reply:
> Why not std::move it back if any errors happen within SendAutoResubscribeRequest?

We could, but again that's not how APIs that take rvalue references normally work in C++.  Normally if you take an rvalue reference it's on you to clean things up correctly.